certification/internal/policy/operator/deployable_by_olm.go

```diff
@@ -63,7 +80,42 @@ func (p *DeployableByOlmCheck) Validate(bundleRef certification.ImageReference)
 		return false, err
 	}
 
-	return p.isCSVReady(installedCSV, *operatorData)
+	p.csvReady, err = p.isCSVReady(installedCSV, *operatorData)
+	if err != nil {
```

Comment:
I think even if the isCSVReady returns an error, we should still run the imageRegistryCheck. Therefore, returning on the line 85 will skip the imageRegistryCheck all together. This means that the return statement on line 96 should be a bit complicated, considering all possible execution outcomes. I do not have an easy solution for this, but I'd be glad to discuss further.

Reply:
Definitely an issue when combining checks like this. In this particular case, I (and I could be mistaken here) see the registry check as dependent on whether the first part succeeds. If it never becomes ready, then checking for the image registry is probably a waste of time, as it may have never even pulled additional images. If you feel strongly about this one, let me know. I'm inclined to leave it as is, for now.